### ISSUE TITLE
FocusTrapZone A11y Pass

### ISSUE THREAD

**Issue by awalia13** [OfficeDev/office-ui-fabric-react#8654]
Tracking all of the issues brought up with the 04/08 accessibility pass on FocusTrapZone control
Details on individual bugs can be found on [Sharepoint](https://microsoft-my.sharepoint.com/:x:/r/personal/v-amaver_microsoft_com/_layouts/15/Doc.aspx?sourcedoc=%7Be4f70d4a-6e63-4d92-b26f-6749296572fb%7D&action=default&uid=%7BE4F70D4A-6E63-4D92-B26F-6749296572FB%7D&ListItemId=1932&ListId=%7B395E1ECB-5002-440E-81C5-6A090E98077C%7D&odsp=1&env=prodbubble)

- ~~Bug1:[Accessibility][Programmatic access]: ARIA attributes must conform to valid values~~
    - JG, 9/19/2019: Following steps in [Sharepoint](https://microsoft-my.sharepoint.com/:x:/r/personal/v-amaver_microsoft_com/_layouts/15/Doc.aspx?sourcedoc=%7Be4f70d4a-6e63-4d92-b26f-6749296572fb%7D&action=default&uid=%7BE4F70D4A-6E63-4D92-B26F-6749296572FB%7D&ListItemId=1932&ListId=%7B395E1ECB-5002-440E-81C5-6A090E98077C%7D&odsp=1&env=prodbubble), bug does not repro on [Fabric 7.35](https://developer.microsoft.com/en-us/fabric#/controls/web/focustrapzone) and Accessibility Insights for Web v2.9.2.
- ~~Bug2:[Accessibility][Programmatic access]: ARIA hidden element contain focusable elements.~~
    - JG, 9/19/2019: Following steps in [Sharepoint](https://microsoft-my.sharepoint.com/:x:/r/personal/v-amaver_microsoft_com/_layouts/15/Doc.aspx?sourcedoc=%7Be4f70d4a-6e63-4d92-b26f-6749296572fb%7D&action=default&uid=%7BE4F70D4A-6E63-4D92-B26F-6749296572FB%7D&ListItemId=1932&ListId=%7B395E1ECB-5002-440E-81C5-6A090E98077C%7D&odsp=1&env=prodbubble), bug does not repro on [Fabric 7.35](https://developer.microsoft.com/en-us/fabric#/controls/web/focustrapzone) and Accessibility Insights for Web v2.9.2.
- ~~Bug3:[Accessibility][Screen Readers]: Edit fields under FocusTrapZone with no tabbable elements does not have labels.~~
    - JG, 9/19/2019: Per [Sharepoint](https://microsoft-my.sharepoint.com/:x:/r/personal/v-amaver_microsoft_com/_layouts/15/Doc.aspx?sourcedoc=%7Be4f70d4a-6e63-4d92-b26f-6749296572fb%7D&action=default&uid=%7BE4F70D4A-6E63-4D92-B26F-6749296572FB%7D&ListItemId=1932&ListId=%7B395E1ECB-5002-440E-81C5-6A090E98077C%7D&odsp=1&env=prodbubble), this is regarding the zero tabbable element example, which is not valid use case for `FocusTrapZone` and has been removed as of [Fabric 7.35](https://developer.microsoft.com/en-us/fabric#/controls/web/focustrapzone).
- ~~Bug4:[Accessibility] [Keyboard Navigation] Focus moves to Microsoft link when the toggle button under Multiple nested FocusTrapZones is turned off.~~
    - JG, 9/19/2019: Following steps in [Sharepoint](https://microsoft-my.sharepoint.com/:x:/r/personal/v-amaver_microsoft_com/_layouts/15/Doc.aspx?sourcedoc=%7Be4f70d4a-6e63-4d92-b26f-6749296572fb%7D&action=default&uid=%7BE4F70D4A-6E63-4D92-B26F-6749296572FB%7D&ListItemId=1932&ListId=%7B395E1ECB-5002-440E-81C5-6A090E98077C%7D&odsp=1&env=prodbubble), bug does not repro on [Fabric 7.35](https://developer.microsoft.com/en-us/fabric#/controls/web/focustrapzone).
- [ ] Bug5:[Functional] When user turns off one of the toggle button of nested trapzone focus should move to the next trapzone instead of moving to Randomize button.
    - JG, 9/20/2019: PR #10556
    - JG, 9/19/2019: [`Randomize` is not a valid scenario for `FocusTrapZone`](https://github.com/OfficeDev/office-ui-fabric-react/blob/676a4717a195fbd83e14738563a0aa2faddfc1b6/packages/office-ui-fabric-react/src/components/FocusTrapZone/examples/FocusTrapZone.Nested.Example.tsx#L86) since it's enabling multiple zones simultaneously, which results in indeterminate behavior. That results in the behavior reported. I will resolve this issue by removing the `Randomize` button in the example.
- ~~Bug6:[Accessibility][Usable]: In scan mode narrator's focus is moving out of the trap zone.~~
    - JG, 9/19/2019: Reopened #6535 to address so I will cross this out as a duplicate.
- ~~Bug7:[Accessibility][Usable]: Screen does not scroll down along with keyboard focus in Backward navigation.~~
    - JG, 9/19/2019: Per [the corresponding video](https://microsoft-my.sharepoint.com/personal/v-amaver_microsoft_com/_layouts/15/onedrive.aspx?id=%2Fpersonal%2Fv%2Damaver%5Fmicrosoft%5Fcom%2FDocuments%2FFabric%20Control%2FP0%20Fabric%20Controls%2FButton%20control%2FBUG%5F6%2Ewmv&parent=%2Fpersonal%2Fv%2Damaver%5Fmicrosoft%5Fcom%2FDocuments%2FFabric%20Control%2FP0%20Fabric%20Controls%2FButton%20control&originalPath=aHR0cHM6Ly9taWNyb3NvZnQtbXkuc2hhcmVwb2ludC5jb20vOnY6L3Avdi1hbWF2ZXIvRWN1b21CUjlwM3hHcmZmN29FeVZpUWNCVFcyR2RCZThfMzJwRl9fTEtFUjlYUT9ydGltZT1xMEl2OENJOTEwZw) this appears to have been an issue with the old documentation website header which has [since been removed](https://developer.microsoft.com/en-us/fabric#/controls/web/focustrapzone).
- ~~Bug8:[Accessibility][Supporting the Platform]: Hyperlinks in the trapzone not adapting to the suggested color in High contrast modes~~
    - JG, 9/19/2019: The [original video](https://microsoft-my.sharepoint.com/personal/v-amaver_microsoft_com/_layouts/15/onedrive.aspx?id=%2Fpersonal%2Fv%2Damaver%5Fmicrosoft%5Fcom%2FDocuments%2FFabric%20Control%2FP2%20Fabric%20Control%2FCallout%2FBug06%2Epng&parent=%2Fpersonal%2Fv%2Damaver%5Fmicrosoft%5Fcom%2FDocuments%2FFabric%20Control%2FP2%20Fabric%20Control%2FCallout&originalPath=aHR0cHM6Ly9taWNyb3NvZnQtbXkuc2hhcmVwb2ludC5jb20vOmk6L3Avdi1hbWF2ZXIvRVJXNmFGaGstcTVDbXlPZ2FuWkUyOUVCSXRXQ3MtQ1FlaUtPandnX1lDNnFJZz9ydGltZT1kVE5UTXlROTEwZw) highlights an example of the issue but it doesn't seem to be an issue with the [`FocusTrapZone` page in Fabric 7.35](https://developer.microsoft.com/en-us/fabric#/controls/web/focustrapzone):
![image](https://user-images.githubusercontent.com/26070760/65265498-f5dfad00-dac5-11e9-87d3-d4945635f4b2.png)

**Comment by joschect** [comment]
@awalia13 thanks for submitting this! We'll add this to our backlog and make sure that the right dev's take a look at it.

**Comment by JasonGore** [comment]
The only actionable bug appears to be Bug5, addressed by #10556. I will tag this issue with Needs: Author Feedback and let it resolve unless additional feedback is provided.
